feat(tests): Add unit tests for web UI Oauth logic

bharos opened the pull request:
<!--
1. Title: [#<issue>] <type>(<scope>): <subject>
   Examples:
     - "[#123] feat(operator): support xxx"
     - "[#233] fix: check null before access result in xxx"
     - "[MINOR] refactor: fix typo in variable name"
     - "[MINOR] docs: fix typo in README"
     - "[#255] test: fix flaky test NameOfTheTest"
   Reference: https://www.conventionalcommits.org/en/v1.0.0/
2. If the PR is unfinished, please mark this PR as draft.
-->

### What changes were proposed in this pull request?
Add some vitest based tests for the OAuth login components for the react JS code

### Why are the changes needed?

These changes were newly added so adding a few tests to make sure things work properly

Fix: #(issue)

### Does this PR introduce _any_ user-facing change?
No
### How was this patch tested?
The tests are ran as Gitlab workflow. The `web-ui-tests` step is passing on the PR

bharos commented:
Hey @LauraXia123 , what do you think about these test changes ? Please review.

Also, not sure if the code changes in pnpm-lock.yaml are expected ?

cc @jerqi

LauraXia123 commented:
@bharos hi, all the UI test is under web/integration-test, can you change it to match this? Thanks!

bharos commented:
> @bharos hi, these files need license header <img alt="image" width="789" height="191" src="https://private-user-images.githubusercontent.com/9210625/480789671-3b2e8dae-c536-473f-82b4-565696c5005e.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU5MDIzMTAsIm5iZiI6MTc1NTkwMjAxMCwicGF0aCI6Ii85MjEwNjI1LzQ4MDc4OTY3MS0zYjJlOGRhZS1jNTM2LTQ3M2YtODJiNC01NjU2OTZjNTAwNWUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDgyMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA4MjJUMjIzMzMwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZTdjN2IwOWVjYTI3ODM1N2EwOTcyNmIzMmQxMTEzYjhjYjNlYzhjYjdiYzAwYzU4YjZlODJlYjI2NzI1ZmM2YyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.kDaEeCa_-oHmqYaIeMbJHhs1bNBnacOsM9J9NV4KFhY">
> 
> And these new introduced libraries need to be listed in the `web/LICENSE.bin`

Thanks. I have fixed the licenses and added headers.
Removed unused dependencies as well. PTAL
